### DISCUSSION TITLE
Possible to exclude models in directory from the generated OpenAPI schema?

### DISCUSSION THREAD

**Discussion by oystein-beaufort** [fastapi/fastapi#8717] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def openapi(self):
  # Can I do something here?
  ...
```

### Description

I want to exclude certain models from the generated OpenAPI schema that are only used internally, ie. models I don't want to be, and that don't need to be visible/public. They are all under the same directory, let's say `/internal/*`. 

Specifically I want to exclude ALL models in `/internal/schema.py` from `components.schemas` in the generated OpenAPI json file.

Is there an easy way to do this?

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9.7

**Comment by insomnes** [comment]
https://fastapi.tiangolo.com/advanced/path-operation-advanced-configuration/exclude-from-openapi

**Comment by oystein-beaufort** [comment]
Thanks, @insomnes, but that's for excluding routes, not models/dirs, so it doesn't help me:(

**Comment by insomnes** [comment]
As i know, you can get raw json, then you can redact it and setup new
```python
raw_schema = app.openapi()
redacted_schema = your_readact_function(raw_schema)
app.openapi_schema = redacted_schema
```

**Comment by oystein-beaufort** [comment]
Thanks again, @insomnes! Yeah, I guess it's possible to do it manually, but I wanted to know if there is an easier solution to this problem.

**Comment by YuriiMotov** [comment, marked as answer]
Don't use this model and it will not be included in openapi.
If you exclude model that is used by your app, you will get broken openapi schema.

You can hide routes or routers that use those models (using `include_in_schema=False`) and those schemas will not appear in openapi.
